DrrImageComputation/Logic/vtkSlicerDrrImageComputationLogic.cxx

```diff
@@ -700,15 +620,30 @@ void vtkSlicerDrrImageComputationLogic::ShowMarkupsNodes(bool toggled)
 }
 
 //----------------------------------------------------------------------------
-vtkMRMLMarkupsClosedCurveNode* vtkSlicerDrrImageComputationLogic::CreateImagerBoundary(vtkMRMLDrrImageComputationNode* parameterNode)
+vtkMRMLMarkupsPlaneNode* vtkSlicerDrrImageComputationLogic::CreateImagerBoundary(vtkMRMLDrrImageComputationNode* parameterNode)
 {
-  vtkNew<vtkMRMLMarkupsClosedCurveNode> imagerMarkupsNode;
-  this->GetMRMLScene()->AddNode(imagerMarkupsNode);
-  imagerMarkupsNode->SetName(IMAGER_BOUNDARY_MARKUPS_NODE_NAME);
-  imagerMarkupsNode->SetCurveTypeToLinear();
+  vtkMRMLScene* scene = this->GetMRMLScene(); 
+  if (!scene)
+  {
+    vtkErrorMacro("CreateImagerBoundary: Invalid MRML scene");
+    return nullptr;
+  }
+  vtkMRMLNode* node = scene->AddNewNodeByClass( "vtkMRMLMarkupsPlaneNode", IMAGER_BOUNDARY_MARKUPS_NODE_NAME);
+
+  vtkMRMLMarkupsPlaneNode* imagerMarkupsNode = nullptr;
+  if (node)
+  {
+    imagerMarkupsNode = vtkMRMLMarkupsPlaneNode::SafeDownCast(node);
```

Comment:
The node doesn't need to be checked twice you can call SafeDownCast and AddNewNodeByClass in the same line.*

Reply:
I will fix that.